rpc/server.go

```diff
@@ -150,20 +150,19 @@ func NewServer(cfg *Config) (*Server, error) {
 		return nil, err
 	}
 
-	reg, err := NewPrometheusRegistry()
-	if err != nil {
-		return nil, err
-	}
-
-	if !isBootnode {
-		SetupMetrics(serverCtx, reg, cfg.Net, cfg.ProtocolBackend, cfg.XMRMaker)
-	}
 	r := mux.NewRouter()
 	r.Handle("/", rpcServer)
 	r.Handle("/ws", wsServer)
+
 	if !isBootnode {
+		reg, err := NewPrometheusRegistry()
+		if err != nil {
+			return nil, err
+		}
+		SetupMetrics(serverCtx, reg, cfg.Net, cfg.ProtocolBackend, cfg.XMRMaker)
```

Comment:
This was just narrowing the scope NewPrometheusRegistry so bootnodes never create it and then not use it.

At the time I was reviewing this code, I was trying to figure out why the integration test bootnode was listening on a random 0.0.0.0 port. It turned out to be a combination of the latest libp2p code and the fact that we didn't configure the bootnode in the integration tests to not connect to the default bootnodes.

Reply:
bootnodes should probably have metrics as well - however yeah in the previous logic, that was not the case. maybe we can re-add in another PR